netaddr.go

```diff
@@ -826,6 +831,37 @@ func (ip IP) appendTo6(ret []byte) []byte {
 	return ret
 }
 
+// StringExpanded is like String but IPv6 addresses are expanded with leading
+// zeroes and no "::" compression. For example, "2001:db8::1" becomes
+// "2001:0db8:0000:0000:0000:0000:0000:0001".
+func (ip IP) StringExpanded() string {
+	switch ip.z {
+	case z0, z4:
+		return ip.String()
+	}
+
+	size := len("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
+	if ip.z != z6noz {
```

Comment:
Try deleting this if statement entirely. When there are zones, it'll allocate twice, but when there aren't, I think it should cut this to a single allocation. (Constant-sized make can go on the stack.)

Reply:
```
10:02 $ go test -bench=BenchmarkIPStringExpanded 
goos: linux
goarch: amd64
pkg: inet.af/netaddr
cpu: AMD Ryzen 9 3900X 12-Core Processor            
BenchmarkIPStringExpanded/v4-24         28673600                48.45 ns/op           16 B/op          1 allocs/op
BenchmarkIPStringExpanded/v6-24         13939674                83.73 ns/op           48 B/op          1 allocs/op
BenchmarkIPStringExpanded/v6_ellipsis-24                13548344                81.87 ns/op           48 B/op          1 allocs/op
BenchmarkIPStringExpanded/v6_v4-24                      13493323                86.92 ns/op           48 B/op          1 allocs/op
BenchmarkIPStringExpanded/v6_zone-24                     8837446               138.5 ns/op           128 B/op          2 allocs/op
PASS
ok      inet.af/netaddr 6.789s
```

@josharian yep that did the trick!